### PR TITLE
V3.6.0 batcher2d mask to native

### PR DESCRIPTION
**Pull request by Zhmz**
Re: #
related PR
https://github.com/cocos/cocos-engine/pull/11592

### Changelog

* mask to native

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/2d/components/mask.ts [hunk 1]
```diff
@@ -365,6 +374,15 @@ export class Mask extends UIRenderer {
         if (this._graphics) {
             this._graphics.onLoad();
         }
+
+        if (JSB) {
+            if (this.renderEntity && this.renderData && this._graphics) {
+                this.renderEntity.setIsMask(true);
```
Comment (dumganhar) [inline cocos/2d/components/mask.ts]: Should be this.`_`renderEntity ?

File: native/cocos/2d/renderer/RenderEntity.h [hunk 2]
```diff
@@ -62,6 +62,15 @@ class RenderEntity final : public Node::UserData {
     void setDynamicRenderDrawInfo(RenderDrawInfo* drawInfo, uint32_t index);
     void removeDynamicRenderDrawInfo();
 
+    inline bool getIsMask() { return _isMask; }
```
Comment (minggo) [inline native/cocos/2d/renderer/RenderEntity.h]: please add const as possible

File: native/cocos/2d/renderer/StencilManager.h [hunk 3]
```diff
@@ -73,19 +74,49 @@ class StencilManager final {
     gfx::DepthStencilState* getDepthStencilState(StencilStage stage, Material* mat = nullptr);
     void setDepthStencilStateFromStage(StencilStage stage);
 
+    inline uint32_t getMaskStackSize() const { return _maskStackSize; }
+    inline void setMaskStackSize(uint32_t size) { _maskStackSize = size; } 
+
+    inline void pushMask() {
+        _maskStackSize++;
```
Comment (minggo) [inline native/cocos/2d/renderer/StencilManager.h]: _maskStackSize++ -> ++_maskStackSize

File: native/cocos/2d/renderer/StencilManager.h [hunk 4]
```diff
@@ -73,19 +74,49 @@ class StencilManager final {
...
+    inline void enableMask() {
+        _stage = StencilStage::ENABLED;
+    }
+
+    inline void exitMask() {
+        if (_maskStackSize == 0) {
+            return;
+        }
+
+        _maskStackSize--;
```
Comment (minggo) [inline native/cocos/2d/renderer/StencilManager.h]: `_maskStackSize--` -> `--_maskStackSize`

File: native/cocos/2d/renderer/StencilManager.h (hunk 3 above)
Comment (Zhmz) [reply native/cocos/2d/renderer/StencilManager.h]: done

File: native/cocos/2d/renderer/StencilManager.h (hunk 4 above)
Comment (Zhmz) [reply native/cocos/2d/renderer/StencilManager.h]: done

**Review by pandamicro — APPROVED** [review]

**Review by SantyWang — APPROVED** [review]

**Pull request merged**